File: Exemplos/App_IntentDigitalHUB_Ionic/Elgin_IntentDigitalHUB_Ionic-font/src/intentDigitalHubService/TERMICA/Commands/AbreConexaoImpressora.ts
import { TermicaCommand } from "../TermicaCommand";

export class AbreConexaoImpressora extends TermicaCommand {
  readonly tipo: number;
  readonly modelo: string;
  readonly conexao: string;
  readonly parametro: number;

  constructor(
    tipo: number,
    modelo: string,
    conexao: string,
    parametro: number
  ) {
    super("AbreConexaoImpressora");
    this.tipo = tipo;
    this.modelo = modelo;
    this.conexao = conexao;
    this.parametro = parametro;
  }

  functionParameters(): object {
    return {
      tipo: this.tipo,
      modelo: this.modelo,
      conexao: this.conexao,
      parametro: this.parametro,
    };
  }
}

File: Exemplos M10/App_IntentDigitalHUB_XamarinAndroid/Elgin_IntentDigitalHUB_XamarinAndroid/IntentServices/Termica/AbreConexaoImpressora.cs
namespace Xamarin_Android_Intent_Digital_Hub.IntentServices.Termica
{
    class AbreConexaoImpressora : TermicaCommand
    {
        readonly int tipo;
        readonly string modelo;
        readonly string conexao;
        readonly int parametro;

        public AbreConexaoImpressora(int tipo, string modelo, string conexao, int parametro) : base("AbreConexaoImpressora")
        {
            this.tipo = tipo;
            this.modelo = modelo;
            this.conexao = conexao;
            this.parametro = parametro;
        }

        protected override string FunctionParameters()
        {
            return "\"tipo\"" + ":" + tipo + "," +
                    "\"modelo\"" + ":" + "\"" + modelo + "\"" + "," +
                    "\"conexao\"" + ":" + "\"" + conexao + "\"" + "," +
                    "\"parametro\"" + ":" + parametro;
        }
    }
}